File: Difficulty: Easy/N meetings in one room/n-meetings-in-one-room.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution
{
    public:
    //Function to find the maximum number of meetings that can
    //be performed in a meeting room.
    int maxMeetings(int start[], int end[], int n)
    {
       vector<pair<int,int>>meetings(n);
       for(int i=0;i<n;i++)
       {
           meetings[i]={end[i],start[i]};
       }
       sort(meetings.begin(),meetings.end());
       
       int cnt=0,lastEndTime=-1;
       
       for(int i=0;i<n;i++)
       {
           if(meetings[i].second > lastEndTime)
           {
               cnt++;
               lastEndTime=meetings[i].first;
           }
       }
       return cnt;
        
    }


};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int start[n], end[n];
        for (int i = 0; i < n; i++) cin >> start[i];

        for (int i = 0; i < n; i++) cin >> end[i];

        Solution ob;
        int ans = ob.maxMeetings(start, end, n);
        cout << ans << endl;
    }
    return 0;
}
// } Driver Code Ends

File: Difficulty: Easy/N meetings in one room/n-meetings-in-one-room.py
#User function Template for python3

class Solution:
    
    #Function to find the maximum number of meetings that can
    #be performed in a meeting room.
    def maximumMeetings(self,start,end):
        # code here
        ls = []
        n = len(start)
        for i in range(n):
            ls.append((start[i], end[i]))
        
        ls.sort(key = lambda x:x[1])
        ans=-1
        count=0
        for s, e in ls:
            if s>ans:
                ans=e
                count+=1
        
        return count

#{ 
 # Driver Code Starts
#Initial Template for Python 3
import atexit
import io
import sys

#Contributed by : Nagendra Jha

if __name__ == '__main__':
    test_cases = int(input())
    for cases in range(test_cases):
        start = list(map(int, input().strip().split()))
        end = list(map(int, input().strip().split()))
        ob = Solution()
        print(ob.maximumMeetings(start, end))
        print("~")

# } Driver Code Ends